### ISSUE TITLE
How To Connect to Existing Stripe Terminal (Or how to delete existing terminal)?

### ISSUE THREAD

**Issue by shesha1096** [stripe/stripe-terminal-android#123]
I'm currently using the Android SDK for the Stripe Terminal. My use case is something like this: So the users of the app make a payment (in-person), for which the app connects to a reader using Bluetooth. For this, I've used the documentation straight up. Everything works fine for the first time, with a successful payment. However, when the user tries to make another payment, the app crashes. The issue was pretty evident, which is, the terminal has already been initialized, and cannot be initialized again. As a workaround, these are some of the approaches that I've already tried:

Connecting to an existing reader - For this, I store the ID of the reader previously, and try connecting to it. When I do this, I get an error saying that the session has expired, and a new token needs to be generated.
I tried using the discoverReaders API to connect to a reader, and I get an exception stating that a new token needs to be created for the terminal transaction. In both the above approaches, it asks me to create a new token before starting the transaction. However, nowhere in the docs was I able to find how to set a new token to an existing terminal.
I tried calling the onConnectReader callback directly, and I had the same issue of using an expired token.
So my question is, how do I set a new token to an existing terminal? If not, can I remove the terminal connection and create a new one? I tried using the clearCachedCredentials() and cleanupTerminal$terminalsdk_release() APIs for this, but none of them worked. Or, is there another way to manage payments when the terminal has already been initialized? For example, calling some other callback directly that handles the payments?

I've been stuck with this issue for the past few days without any outcome. It'd be great to get some sort of closure on this. Thank you!

**Comment by billfinn-stripe** [comment]
Hi @shesha1096 -- after connecting to a reader, you should be able to accept multiple payments using the same reader without having to disconnect and reconnect. And, as you point out, you should not call `initTerminal` more than once.

I'm going to optimistically close this issue, but if you are still having problems, please follow up with a minimal reproducible example.

**Comment by ajithv11** [comment]
@billfinn-stripe  OK but at some point I got the following error When I select that to test and go to donate it gives an error message stating NFC reader connection the Stripe Terminal ConnectionToken has expired. Please download a new one using v1/terminal/connection_tokens

So I think I should call initTerminal on each payment, Unfortunately I couldn't able to call initTerminal multiple times , here I struck with the following error 
_You can only call initTerminal *before* requesting the Terminal instance for the first time. If you are trying to switch accounts in your app, refer to the documentation for the `clearCachedCredentials` method._

Without initTerminal, How can I generalte new token v1/terminal/connection_tokens

**Comment by billfinn-stripe** [comment]
> While running the tap to pay on my mobile device with NFC support, at some point I got the following error message stating NFC reader connection the Stripe Terminal ConnectionToken has expired. Please download a new one using v1/terminal/connection_tokens

Can you create a new issue, fill out the template, and include a stacktrace where you are seeing this error?

**Comment by udayTechnosouls** [comment]
@billfinn-stripe 
I have connect one M2 device and try to connect second device but it is show me token error like "The
                                         **Stripe Terminal ConnectionToken has already been redeemed.** Please generate a new one using
                                         v1/terminal/connection_tokens"
                                         
request :- class=RestClient message="url=https://api.stripe.com/v1/terminal/connection_tokens/activate
                                         request={"connectionType":"bluetooth","deviceSerialNumber":"STRM26146007039","deviceType":"stripe_m2","posDeviceId":"fa00ceeb-3e50-4640-9cfa-7fd2b59d3cc8","posDeviceInfo":{"appModel":{"appId":"com.membershiptoolkit.mobileapp","appVersion":"2.2.4"},"deviceClass":"POS","deviceUuid":"fa00ceeb-3e50-4640-9cfa-7fd2b59d3cc8","hardwareModel":{"posInfo":{"description":"SM-A226B"}},"hostHwVersion":"SM-A226B","hostOsVersion":"13"},"posVersionInfo":{"clientType":"ANDROID_SDK","clientVersion":"2.19.0"},"reader":{"appModel":{"appId":"firmware-config-key","appVersion":"2.01.00.20-SZZZ_Prod_US_v5-480001"},"bbposBaseConfig":"SZZZ_Prod_US_v5","bbposFirmware":"2.01.00.20","bbposKeyProfile":"480001","deviceClass":"READER","deviceUuid":"STRM26146007039","hardwareModel":{"bbposHardware":"STRIPEM2"},"hostHwVersion":"STRM2-01
                                         (0803)"},"registerToLocation":"tml_FHwmfgXKcv8DZY","reportedReaderConfig":{"keyId":"480001"}}"

**Comment by billfinn-stripe** [comment]
I see you filed https://github.com/stripe/stripe-terminal-android/issues/425. Leaving this issue closed.
